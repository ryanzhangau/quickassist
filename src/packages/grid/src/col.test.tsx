import React from 'react'
import { render } from '@testing-library/react'
import AnctechCols, { generateClass } from './col'
import { IDictionary, VAlignment } from './index'

describe('AnctechCols component', () => {
  it('should match <AnctechCols /> snapshot', () => {
    const { asFragment } = render(<AnctechCols>test</AnctechCols>)
    expect(asFragment()).toMatchSnapshot()
  })

  it('should generate col-* classes when colSize is number', () => {
    const colSize = 3
    expect(generateClass({ colSize })).toBe('col-3')
  })

  it('should generate col-<screenSize>-* classes when colSize is object', () => {
    const colSize: IDictionary<string | number> = { xs: 3, sm: 4, lg: 'auto' }
    expect(generateClass({ colSize })).toBe(' col-xs-3 col-sm-4 col-lg-auto')
  })

  it('should generate offset-<screenSize>-* classes when offset is provided', () => {
    const offset: IDictionary<string | number> = { xs: 3, sm: 4, lg: 'auto' }
    expect(generateClass({ offset })).toBe('col offset-xs-3 offset-sm-4 offset-lg-auto')
  })

  it('should generate align-self-* classes when selfAlign is provided', () => {
    const selfAlign: VAlignment = 'start'
    expect(generateClass({ selfAlign })).toBe('col align-self-start')
  })
})
