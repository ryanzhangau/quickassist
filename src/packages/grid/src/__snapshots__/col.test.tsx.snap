// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnctechCols component should match <AnctechCols /> snapshot 1`] = `
<DocumentFragment>
  <div
    class="col"
  >
    test
  </div>
</DocumentFragment>
`;
